src/panels/lovelace/cards/hui-map-card.ts

```diff
@@ -309,46 +331,49 @@ class HuiMapCard extends LitElement implements LovelaceCard {
     return color;
   }
 
-  private _getEntities = memoizeOne(
-    (
-      states: HassEntities,
-      config: MapCardConfig,
-      configEntities?: MapEntityConfig[]
-    ) => {
-      if (!states || !config) {
-        return undefined;
-      }
+  private _getSourceEntities(
+    states?: HassEntities,
+    config?: MapCardConfig
+  ): string[] {
+    if (!states || !config || !config.geo_location_sources) {
+      return [];
+    }
 
-      const geoEntities: string[] = [];
-      if (config.geo_location_sources) {
-        // Calculate visible geo location sources
-        const includesAll = config.geo_location_sources.includes("all");
-        for (const stateObj of Object.values(states)) {
-          if (
-            computeDomain(stateObj.entity_id) === "geo_location" &&
-            (includesAll ||
-              config.geo_location_sources.includes(stateObj.attributes.source))
-          ) {
-            geoEntities.push(stateObj.entity_id);
-          }
-        }
+    const geoEntities: string[] = [];
+    // Calculate visible geo location sources
+    const includesAll = config.geo_location_sources.includes("all");
+    for (const stateObj of Object.values(states)) {
+      if (
+        computeDomain(stateObj.entity_id) === "geo_location" &&
+        (includesAll ||
+          config.geo_location_sources.includes(stateObj.attributes.source))
+      ) {
+        geoEntities.push(stateObj.entity_id);
       }
+    }
+    return geoEntities;
+  }
 
-      return [
-        ...(configEntities || []).map((entityConf) => ({
-          entity_id: entityConf.entity,
-          color: this._getColor(entityConf.entity),
-          label_mode: entityConf.label_mode,
-          focus: entityConf.focus,
-          name: entityConf.name,
-        })),
-        ...geoEntities.map((entity) => ({
+  private _updateMapEntities(): void {
+    const entities = [
+      ...(this._configEntities || []).map((entityConf) => ({
+        entity_id: entityConf.entity,
+        color: this._getColor(entityConf.entity),
+        label_mode: entityConf.label_mode,
+        focus: entityConf.focus,
+        name: entityConf.name,
+      })),
+      ...this._getSourceEntities(this.hass?.states, this._config).map(
+        (entity) => ({
           entity_id: entity,
           color: this._getColor(entity),
-        })),
-      ];
+        })
+      ),
+    ];
+    if (!deepEqual(entities, this._mapEntities)) {
```

Comment:
If we move the check from `shouldUpdate` to `willUpdate`, we can remove this.

Reply:
Done